libconcurrent/includes/sim.h

```diff
@@ -42,10 +42,10 @@ typedef struct SimObjectState {
 
 typedef union pointer_t {
     struct StructData {
-        int64_t seq : 32;
-        int32_t index : 32;
+        uint64_t seq : 32;
+        uint32_t index : 32;
```

Comment:
this is always 32 bits so why the extra `: 32`?

Reply:
The sequence number increases one time per operation at most. 
I think, 32 bits for a sequence number is too low. Thus, I'm giving 48 bits and the remaining bits (i.e. 24) are enough for indexing arrays in such a data-structure. 
Recall that the total size of this struct should be 64 bits in order to be atomic to most modern machine architectures.
Thus the new structure will be: 
struct StructData {
        uint64_t seq : 48;
        uint32_t index : 24;
    } struct_data;

Reply:
Actually, it should be 24 and 40!